### PR TITLE
Bug 1120733 - [Privacy Panel] The Back arrow button disappears after doi...

### PR DESCRIPTION
**Pull request by mancas**
...ng some actions

### CODE REVIEW

**Comment by try-server-hook** [comment]
[mancas Manuel Casas (mancas)](https://github.com/mancas) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=8cf9b3d390b8)

File: apps/privacy-panel/js/root/main.js [hunk 1]
```diff
@@ -28,6 +28,7 @@ function(panels, SettingsListener, SettingsHelper) {
 
       this.panel = document.getElementById('root');
       this.backBtn = this.panel.querySelector('#back-to-settings');
+      this.fromSettings = null;
```
Comment (EragonJ) [inline apps/privacy-panel/js/root/main.js]: by default, `this.fromSettings` would be `false`. (Make its type consistent would be more understandable)

File: apps/privacy-panel/js/root/main.js [hunk 2]
```diff
@@ -43,8 +44,16 @@ function(panels, SettingsListener, SettingsHelper) {
 
       // Reset launch flag when app is not active.
       window.addEventListener('blur', function() {
+        this.fromSettings = this.panel.dataset.settings;
         SettingsHelper('privacypanel.launched.by.settings').set(false);
-      });
+      }.bind(this));
```
Comment (EragonJ) [inline apps/privacy-panel/js/root/main.js]: you can use arrow function and by doing so, you don't have to `bind(this)` anymore.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions

File: apps/privacy-panel/js/root/main.js [hunk 3]
```diff
@@ -43,8 +44,16 @@ function(panels, SettingsListener, SettingsHelper) {
...
+
+      // Restore launch flag when app is active again
+      window.addEventListener('focus', function() {
+        if (this.fromSettings) {
+          SettingsHelper('privacypanel.launched.by.settings').set(true);
+        }
+      }.bind(this));
```
Comment (EragonJ) [inline apps/privacy-panel/js/root/main.js]: same here !

**Comment by try-server-hook** [comment]
[mancas Manuel Casas (mancas)](https://github.com/mancas) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=32f150f08d1e)

**Comment by try-server-hook** [comment]
[mancas Manuel Casas (mancas)](https://github.com/mancas) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=5dcbf589147f)

File: apps/privacy-panel/js/root/main.js [hunk 4]
```diff
@@ -34,16 +32,14 @@ function(panels, SettingsListener, SettingsHelper) {
...
       this.events();
     },
 
     events: function() {
       panels.registerEvents([this.panel]);
 
-      // Reset launch flag when app is not active.
-      window.addEventListener('blur', function() {
-        SettingsHelper('privacypanel.launched.by.settings').set(false);
+      window.addEventListener('visibilitychange', function(evt) {
```
Comment (EragonJ) [inline apps/privacy-panel/js/root/main.js]: `evt` is not used, so we can remove it !

File: apps/privacy-panel/style/panels.css [hunk 5]
```diff
@@ -10,20 +10,6 @@
...
-
-section[data-settings="false"] #back-to-settings {
-  visibility: hidden;
-}
-
-section[data-settings="true"] #back-to-settings {
-  visibility: visible;
-}
-
-
```
Comment (EragonJ) [inline apps/privacy-panel/style/panels.css]: yeah !

**Comment by try-server-hook** [comment]
[mancas Manuel Casas (mancas)](https://github.com/mancas) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=5edde7b57d45)

**Comment by try-server-hook** [comment]
[mancas Manuel Casas (mancas)](https://github.com/mancas) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=caa5129cf6c0)

**Comment by try-server-hook** [comment]
[mancas Manuel Casas (mancas)](https://github.com/mancas) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=19aaa2a9623c)

**Comment by try-server-hook** [comment]
[mancas Manuel Casas (mancas)](https://github.com/mancas) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=1f335e287994)

**Comment by try-server-hook** [comment]
[mancas Manuel Casas (mancas)](https://github.com/mancas) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=33521f732748)

**Comment by try-server-hook** [comment]
[mancas Manuel Casas (mancas)](https://github.com/mancas) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=2010ce93d3af)

**Pull request merged**